contingency/contingency-api/src/main/java/com/powsybl/contingency/json/ContingencyListDeserializer.java

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright (c) 2022, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.contingency.json;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+import com.powsybl.contingency.ContingencyList;
+
+import java.io.IOException;
+
+/**
+ * @author Etienne Lesot <etienne.lesot@rte-france.com>
+ */
+public class ContingencyListDeserializer extends StdDeserializer<ContingencyList> {
+
+    public ContingencyListDeserializer() {
+        super(ContingencyList.class);
+    }
+
+    @Override
+    public ContingencyList deserialize(JsonParser parser, DeserializationContext deserializationContext) throws IOException {
+        parser.nextToken();
+        while (parser.getCurrentName() != null) {
+            if ("type".equals(parser.getCurrentName())) {
+                switch (parser.nextTextValue()) {
+                    case "default":
```

Comment:
We should probably try to leverage jackson mechanisms to automate the association name -> deserializer.
Otherwise we change a little the behaviour of the deserializers used below, because it will expect to have already parsed the beginning of the object, so if someone wants to use them to deserialize a plain "DefaultContingencyList" for example, it will not work.

If we want to customize this, I think the only way is to use the `JsonTypeInfo` annotation, and it's probably slightly better on a "mixin" object to not define it directly into the `ContingencyList` interface.